viewing_party/party.py

```diff
@@ -1,23 +1,169 @@
+        if title not in friends_movies[1]["watched"] and title not in friends_movies[2]["watched"]:
+            users_unique_movies.append(title)
+    return users_unique_movies
+
+def get_friends_unique_watched(user_data):
+    friends_unique_movies = []
+    friends_movies = create_friends_dictionary(user_data)
+    
+    for movie in friends_movies[1]["watched"]:
+        if movie not in user_data["watched"]:
+            if movie not in friends_unique_movies:
+                friends_unique_movies.append(movie)
+    for friend_movies in friends_movies[2]["watched"]:
+        if friend_movies not in user_data["watched"]:
+            if friend_movies not in friends_unique_movies:
+                friends_unique_movies.append(friend_movies)
+    return friends_unique_movies
+
 # -----------------------------------------
 # ------------- WAVE 4 --------------------
 # -----------------------------------------
+def get_available_recs(user_data):
+    recommendations = []
+    friends_movies = create_friends_dictionary(user_data)
+
+    for movie in friends_movies[1]["watched"]:
+        if movie not in user_data["watched"] and movie["host"] in user_data["subscriptions"]:
+            recommendations.append(movie)
+        for movie in friends_movies[2]["watched"]:
+            if movie not in user_data["watched"] and movie["host"] in user_data["subscriptions"]:
+                recommendations.append(movie)
+    return recommendations
 
 # -----------------------------------------
 # ------------- WAVE 5 --------------------
 # -----------------------------------------
 
+def get_new_rec_by_genre(user_data):
+    recommendations = []
+    friends_movies = create_friends_dictionary(user_data)
+
+    if user_data["watched"] != []:
+        if friends_movies[1]["watched"] != []:
+            for movie in friends_movies[1]["watched"]:
+                if movie not in user_data["watched"] and movie["host"] in user_data["subscriptions"]:
+                    recommendations.append(movie)
+            if friends_movies[2]["watched"] != []:
+                for movie in friends_movies[2]["watched"]:
+                    if movie not in user_data["watched"] and movie["host"] in user_data["subscriptions"]:
+                        recommendations.append(movie)
+            for rec in recommendations:
+                if rec["genre"] not in user_data["watched"][0]["genre"]:
+                    recommendations.remove(rec)
+    return recommendations
+
+def get_rec_from_favorites(user_data):
+    recommendations = []
+    friends_movies = create_friends_dictionary(user_data)
+
+    for movie in user_data["watched"]:
+        if user_data["favorites"] != []:
+            if friends_movies[1]["watched"] != [] and friends_movies[2]["watched"] != []:
+                if movie in user_data["favorites"] and movie not in friends_movies[1]["watched"] and movie not in friends_movies[2]["watched"] and movie not in recommendations:

```

Comment:
This considers all the requirements for the favorite recommendation, but what if there weren't always 2 friends?